taxi/views.py

```diff
@@ -1,47 +1,62 @@
 from django.shortcuts import render
 from django.views import generic
+from django.views.generic import TemplateView, ListView, DetailView
+from django.contrib.auth.mixins import LoginRequiredMixin
+from pip._internal.utils import logging
+from .models import Car, Driver, Manufacturer
 
-from .models import Driver, Car, Manufacturer
 
-
-def index(request):
-    """View function for the home page of the site."""
-
-    num_drivers = Driver.objects.count()
-    num_cars = Car.objects.count()
-    num_manufacturers = Manufacturer.objects.count()
-
-    context = {
-        "num_drivers": num_drivers,
-        "num_cars": num_cars,
-        "num_manufacturers": num_manufacturers,
-    }
-
-    return render(request, "taxi/index.html", context=context)
-
-
-class ManufacturerListView(generic.ListView):
+class ManufacturerListView(LoginRequiredMixin, generic.ListView):
     model = Manufacturer
     context_object_name = "manufacturer_list"
     template_name = "taxi/manufacturer_list.html"
     paginate_by = 5
 
 
-class CarListView(generic.ListView):
+class CarListView(LoginRequiredMixin, generic.ListView):
     model = Car
     paginate_by = 5
     queryset = Car.objects.select_related("manufacturer")
+    template_name = "taxi/car_list.html"
 
 
-class CarDetailView(generic.DetailView):
+class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
+    template_name = "taxi/car_detail.html"
 
 
-class DriverListView(generic.ListView):
+class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
-    paginate_by = 5
+    queryset = Driver.objects.prefetch_related("cars__manufacturer")
+    template_name = "taxi/driver_detail.html"
+
 
+logger = logging.getLogger(__name__)
```

Comment:
please remove logging from here 
or if you want to use logging then do it properly and use it for the whole project